TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/controller/BotControllerActivity.java

```diff
@@ -0,0 +1,65 @@
+package com.novoda.tpbot.controller;
+
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+
+import com.novoda.notils.caster.Views;
+import com.novoda.tpbot.R;
+import com.novoda.tpbot.SelfDestructingMessageView;
+
+public class BotControllerActivity extends AppCompatActivity {
+
+    private static final String LAZERS = String.valueOf(Character.toChars(0x1F4A5));
+
+    private static final long DURATION_DIRECTIONS = 1500L;
+    private static final long DURATION_LAZERS = 150L;
+
+    private SelfDestructingMessageView debugView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_bot_controller);
+        debugView = Views.findById(this, R.id.bot_controller_debug_view);
+
+        DirectionPadView padView = Views.findById(this, R.id.bot_controller_direction_view);
+        padView.setOnDirectionPressedListener(new OnDirectionPressedListener() {
+
+            @Override
+            public void onDirectionPressed(BotDirection direction) {
+                String arrowCharacter = arrowOf(direction);
+                debugView.showPermanently(arrowCharacter);
+            }
+
+            @Override
+            public void onDirectionReleased(BotDirection direction) {
+                String arrowCharacter = arrowOf(direction);
+                debugView.showTimed(arrowCharacter + " released", DURATION_DIRECTIONS);
+            }
+
+            private String arrowOf(BotDirection direction) {
+                switch (direction) {
```

Comment:
I'd suggest getting rid of the switch case statement since it is considered a code smells 👃. You perhaps could instantiate the enum with a value and use the direction parameter to get it. This [link](https://sourcemaking.com/refactoring/smells/switch-statements) here could be useful as a reference of treatments for this kind of statement.

Another solution that I'd use in swift, could be having a array of directions and assigned values(string) and then get the direction out of the array since your enum is defined and can't return a non treated type.